pkg/provisioningv2/rke2/planner/encryptionkeyrotation.go

```diff
@@ -700,31 +762,38 @@ func encryptionKeyRotationWaitForSecretsEncryptStatus(cp *rkev1.RKEControlPlane)
 // Since reencrypt can be any of three statuses (reencrypt_request, reencrypt_active, and reencrypt_finished), any of these stages are valid, however the first two (request & active) do
 // not explicitly indicate that the current command was successful, just that it hasn't failed yet. In certain cases, a
 // cluster may never move beyond request and active.
-func encryptionKeyRotationIsCurrentStageAllowed(leaderStage string, currentPhase rkev1.RotateEncryptionKeysPhase) bool {
+func encryptionKeyRotationIsCurrentStageAllowed(leaderStage string, currentPhase rkev1.RotateEncryptionKeysPhase) error {
 	switch currentPhase {
 	case rkev1.RotateEncryptionKeysPhasePrepare:
-		return leaderStage == encryptionKeyRotationCommandPrepare
+		if leaderStage == encryptionKeyRotationCommandPrepare {
+			return nil
+		}
 	case rkev1.RotateEncryptionKeysPhaseRotate:
-		return leaderStage == encryptionKeyRotationCommandRotate
+		if leaderStage == encryptionKeyRotationCommandRotate {
+			return nil
+		}
 	case rkev1.RotateEncryptionKeysPhaseReencrypt:
 		if leaderStage == encryptionKeyRotationStageReencryptRequest ||
 			leaderStage == encryptionKeyRotationStageReencryptActive ||
 			leaderStage == encryptionKeyRotationStageReencryptFinished {
-			return true
+			return nil
 		}
-		return false
 	}
-	return false
+	return fmt.Errorf("unexpected encryption key rotation stage [%s] for phase [%s]", leaderStage, currentPhase)
 }
 
 // encryptionKeyRotationFailed updates the various status objects on the control plane, allowing the cluster to
 // continue the reconciliation loop. Encryption key rotation will not be restarted again until requested.
-func (p *Planner) encryptionKeyRotationFailed(cp *rkev1.RKEControlPlane) error {
-	err := p.setEncryptionKeyRotateState(cp, cp.Spec.RotateEncryptionKeys, rkev1.RotateEncryptionKeysPhaseFailed)
-	if err != nil {
+func (p *Planner) encryptionKeyRotationFailed(cp *rkev1.RKEControlPlane, err error) error {
+	if err := p.setEncryptionKeyRotateState(cp, cp.Spec.RotateEncryptionKeys, rkev1.RotateEncryptionKeysPhaseFailed); err != nil {
```

Comment:
It looks like the value of the parameter `err` passed into the function will be overwritten by the local variable `err`, is it expected? Or am I misunderstanding anything?

Reply:
I used `if err :=` to purposefully shadow, so it isn't being overwritten.

Reply:
So the scope of the defined err here is only inside if and the err used further below is the one passed to func, correct?
if that’s the case I’d rather chose a different name to avoid confusion rather than add a comment.

Reply:
@jakefhyde  Can you update your PR based on the above comment?